src/adapter/search-request-adapter/search-params-adapter.ts

```diff
@@ -59,13 +59,27 @@ export function adaptSearchParams(
 
   const placeholderSearch = searchContext.placeholderSearch
   const query = searchContext.query
-  const paginationTotalHits = searchContext.paginationTotalHits
 
-  // Limit
-  if ((!placeholderSearch && query === '') || paginationTotalHits === 0) {
+  // Pagination
+  const { pagination } = searchContext
```

Comment:
> const meiliSearchParams: Record<string, any> = {}, shouldn't be const meiliSearchParams: MeiliSearchParams = {}?

The issue lies with `InstantSearchParams` being read-only and `MeiliSearchParams` being not read-only. It results for example in 
```
meiliSearchParams.facetsDistribution = searchContext?.facets
```
throwing: 
```
Type 'readonly string[] | undefined' is not assignable to type 'string[] | undefined'.
  The type 'readonly string[]' is 'readonly' and cannot be assigned to the mutable type 'string[]'.t
```

But by saying `Record<string, any>`  it removes this conflict. At the end, I still return a `MeiliSearchParams` so if I created the object incorrectly it will throw an error anyway.